### PR TITLE
[ST] NetworkPoliciesST refactor and merge of several tests into improved Network policies test

### PR DESCRIPTION
**Pull request by henryZrncik**
### Type of change

- Refactoring

### Description

`NetworkPoliciesST` merging of tests following tests into the one: 
- `testNetworkPoliciesWithPlainListener`
- `testNetworkPoliciesWithTlsListener`
- `testNPWhenOperatorIsInSameNamespaceAsOperand` 

now both network policies are set for one kafka cluster and several clients (plain listener, tls listener, with and without suporitng network policy). 

### Checklist

- [x] Write tests
- [x] Make sure all tests pass
- [x] Update documentation

### CODE REVIEW

File: systemtest/src/main/java/io/strimzi/systemtest/utils/ClientUtils.java [hunk 1]
```diff
@@ -214,5 +216,15 @@ public static String generateRandomConsumerGroup() {
 
         return CONSUMER_GROUP_NAME + salt;
     }
+
+    public static KafkaClientsBuilder getDummyClientBuilder(TestStorage testStorage) {
```
Comment (im-konge) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/ClientUtils.java:220]:
```suggestion
    public static KafkaClientsBuilder getDefaultClientBuilder(TestStorage testStorage) {
```
maybe?

File: systemtest/src/main/java/io/strimzi/systemtest/utils/ClientUtils.java [hunk 2]
```diff
@@ -214,5 +216,15 @@ public static String generateRandomConsumerGroup() {
...
+        return  new KafkaClientsBuilder()
```
Comment (im-konge) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/ClientUtils.java:221]:
```suggestion
        return new KafkaClientsBuilder()
```

File: systemtest/src/test/java/io/strimzi/systemtest/security/NetworkPoliciesST.java [hunk 3]
```diff
@@ -189,67 +143,99 @@ void testNetworkPoliciesWithTlsListener(ExtensionContext extensionContext) {
...
+            KafkaTopicTemplates.topic(testStorage.getClusterName(), topicNameAccessedPlain, testStorage.getNamespaceName()).build(),
+            KafkaTopicTemplates.topic(testStorage.getClusterName(), topicNameAccessedTls, testStorage.getNamespaceName()).build(),
+            KafkaTopicTemplates.topic(testStorage.getClusterName(), topicNameDeniedPlain, testStorage.getNamespaceName()).build(),
+            KafkaTopicTemplates.topic(testStorage.getClusterName(), topicNameDeniedTls, testStorage.getNamespaceName()).build()
         );
 
-        LOGGER.info("Verifying that producer/consumer: {}/{} are able to exchange messages", testStorage.getProducerName(), testStorage.getConsumerName());
+        LOGGER.info("Initialize producers and consumers with access to the Kafka using plain and tls listeners");
+        KafkaClients kafkaClientsWithAccessPlain = ClientUtils.getDummyClientBuilder(testStorage)
+            .withBootstrapAddress(KafkaResources.plainBootstrapAddress(testStorage.getClusterName()))
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/security/NetworkPoliciesST.java:172]: You have `plainBootstrapAddress` already in the clients method, so you don't need to specify it here (just a nit)

**Review by im-konge — APPROVED** [review]
Just nits, thanks :)

File: systemtest/src/test/java/io/strimzi/systemtest/security/NetworkPoliciesST.java (hunk 3 above)
Comment (henryZrncik) [reply systemtest/src/test/java/io/strimzi/systemtest/security/NetworkPoliciesST.java:172]: sure, makes sense 👍

**Comment by henryZrncik** [comment]
@strimzi-ci run tests --profile=all --install-type=bundle --cluster-type=ocp --cluster-version=4.14 --testcase=NetworkPoliciesST

**Review by see-quick — APPROVED** [review]

**Pull request merged**